patroni/ha.py

```diff
@@ -909,6 +918,15 @@ def _is_healthiest_node(self, members: Collection[Member], check_replication_lag
                     if not self.sync_mode_is_active() or not self.cluster.sync.leader_matches(st.member.name):
                         return False
                     logger.info('Ignoring the former leader being ahead of us')
+                if my_wal_position == st.wal_position and self.patroni.failover_priority < st.failover_priority:
+                    # There's a higher priority non-lagging replica
+                    logger.info(
+                        '%s is not lagging and has priority %s, while this node has priority %s',
```

Comment:
I guess "not lagging" is not an accurate message here.
The node can be considered the best candidate for a promotion even if it's lagging, as long as `maximum_lag_on_failover` is respected.
I'd say that although it's unlikely that two or more candidate nodes are lagging with the very same amount of lag, it is still possible.
Thoughts?

Reply:
This makes sense! This I believe is a holdover from an earlier version that would select for priority among all nodes that were within `maximum_lag_on_failover`. What about:

`'%s has equally tolerable WAL position and has priority %s, while this node has priority %s',`

Reply:
I think it looks better, yes!